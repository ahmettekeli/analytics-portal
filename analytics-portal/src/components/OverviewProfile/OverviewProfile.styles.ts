import styled from "styled-components";
import { colors } from "../../constants";

export const Wrapper = styled.div`
  display: flex;
  justify-content: left;
  margin: 1rem;
  padding: 0.5rem;
  min-width: 100%;
  min-height: 10%;
`;

export const ProfileItem = styled.div`
  display: flex;
  flex-direction: column;
  margin: 0 1rem;

  p {
    color: ${colors.dateText};
    font-size: 0.6rem;
    font-weight: bold;
    padding: 1rem 0;
  }

  img {
    width: 8rem;
    height: 8rem;
  }
`;

export const NameContainer = styled.div`
  font-size: 1.5rem;
  font-weight: bold;
  text-align: left;
`;
