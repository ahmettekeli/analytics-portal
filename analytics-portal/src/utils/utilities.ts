import { v4 as uuidv4 } from "uuid";
import {
  AppDataInterface,
  AppChartDataType,
  CampaignChartDataType,
  CampaignType,
} from "../context/Interfaces";

const dummyCampaignData = [
  { day: "Day 1" },
  { day: "Day 2" },
  { day: "Day 3" },
  { day: "Day 4" },
  { day: "Day 5" },
  { day: "Day 6" },
  { day: "Day 7" },
];

export function isNameValid(name: string): boolean {
  return name.length > 0;
}

export function getAvgRevenue(item: AppDataInterface, length: number): number {
  let avgRevenue: number =
    item.revenue.reduce((acc, rev) => acc + rev.value, 0) / length;
  return Number(avgRevenue.toFixed(2));
}

export function getAvgInstalls(item: AppDataInterface, length: number): number {
  let avgInstalls: number =
    item.installs.reduce((acc, ins) => acc + ins.value, 0) / length;
  return Number(avgInstalls.toFixed(2));
}

export function formatDate(date: Date): string {
  let dateArr = date.toString().split("T")[0].split("-");
  return `${dateArr[2]}.${dateArr[1]}.${dateArr[0]}`;
}

export function getRandomValue(min: number, max: number): number {
  return Math.trunc(Math.random() * (max - min) + min);
}

export function generateCampaign(name: string): CampaignType {
  let installs = dummyCampaignData.map((item) => {
    return {
      day: item.day,
      value: getRandomValue(0, 200),
    };
  });

  return {
    name,
    id: uuidv4(),
    installs,
  };
}

export function getAppChartData(app: AppDataInterface): AppChartDataType {
  return {
    installLabels: app.installs.map((item) => item.day),
    installData: app.installs.map((item) => item.value),
    revenueLabels: app.revenue.map((item) => item.day),
    revenueData: app.revenue.map((item) => item.value),
  };
}

export function getCampaignChartData(
  campaign: CampaignType
): CampaignChartDataType {
  return {
    installLabels: campaign.installs.map((item) => item.day),
    installData: campaign.installs.map((item) => item.value),
  };
}

export function isCampaignPathValid(name: string, appName: string): boolean {
  return name === appName;
}
