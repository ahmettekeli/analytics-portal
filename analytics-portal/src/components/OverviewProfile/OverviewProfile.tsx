import * as S from "./OverviewProfile.styles";
import capitalize from "lodash/capitalize";

function OverviewProfile({
  imgUrl,
  name,
  creationDate,
}: {
  imgUrl: string | undefined;
  name: string | undefined;
  creationDate: string;
}) {
  return (
    <S.Wrapper data-testid="OverviewProfile">
      <S.ProfileItem>
        <S.ProfileImg src={imgUrl} alt={name} />
      </S.ProfileItem>
      <S.ProfileItem>
        <S.NameContainer>{capitalize(name)}</S.NameContainer>
        <S.ProfileText>{`Created at ${creationDate}`}</S.ProfileText>
      </S.ProfileItem>
    </S.Wrapper>
  );
}

export default OverviewProfile;
