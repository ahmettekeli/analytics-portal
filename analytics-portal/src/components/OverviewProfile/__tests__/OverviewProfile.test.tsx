import { cleanup, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import OverviewProfile from "../OverviewProfile";
import { BrowserRouter as Router } from "react-router-dom";
import { formatDate } from "utils/utilities";
import { after } from "lodash";

describe("OverviewProfile", () => {
  const imgUrl = "http://placeimg.com/640/480/abstract";
  const name = "Test";
  const creationDate = new Date("2021-11-03T12:33:53.023Z");
  let renderResult: ReturnType<typeof render>;

  beforeEach(() => {
    renderResult = render(
      <Router>
        <OverviewProfile
          imgUrl={imgUrl}
          name={name}
          creationDate={formatDate(creationDate)}
        />
      </Router>
    );
  });

  afterEach(cleanup);

  test("Renders OverviewProfile component", () => {
    expect(renderResult.getByTestId("OverviewProfile")).toBeInTheDocument();
  });

  test("Has correct profile image", () => {
    expect(
      renderResult.getByTestId("OverviewProfile").getElementsByTagName("img")[0]
        .src
    ).toBe(imgUrl);
  });

  test("Has correct name", () => {});
  //TODO
  test("Has correct created at date", () => {});
  //TODO
  test("OverviewProfile matches snapshot", () => {
    const tree = renderer
      .create(
        <Router>
          <OverviewProfile
            imgUrl={imgUrl}
            name={name}
            creationDate={formatDate(creationDate)}
          />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
