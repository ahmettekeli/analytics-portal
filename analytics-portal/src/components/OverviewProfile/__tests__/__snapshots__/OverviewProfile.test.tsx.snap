// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewProfile OverviewProfile matches snapshot 1`] = `
<div
  className="sc-bdvvtL dfeucS"
  data-testid="OverviewProfile"
>
  <div
    className="sc-gsDKAQ kpcoSg"
  >
    <img
      alt="Test"
      className="sc-hKwDye eprGRg"
      src="http://placeimg.com/640/480/abstract"
    />
  </div>
  <div
    className="sc-gsDKAQ kpcoSg"
  >
    <div
      className="sc-eCImPb hRZqSQ"
    >
      Test
    </div>
    <p
      className="sc-dkPtRN iMrBjb"
    >
      Created at undefined.undefined.Wed Nov 03 2021 15:33:53 GM
    </p>
  </div>
</div>
`;
